### PR TITLE
#2484 Restore workspace if the upgrade failed

### PR DESCRIPTION
**Pull request by wu87988622**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #2484 )
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/Plugins/Plugins.js [hunk 1]
```diff
@@ -39,26 +40,34 @@ const Plugins = props => {
...
   );
-  const { getData: pipelineRes, getApi: getPipeline } = useApi.useGetApi(
-    URL.PIPELINE_URL,
-  );
-  const { putApi: putWorker } = useApi.usePutApi(URL.WORKER_URL);
-  const { putApi: putConnector } = useApi.usePutApi(URL.CONNECTOR_URL);
-  const { waitApi } = useApi.useWaitApi();
   const { showMessage } = useSnackbar();
+  const {
+    getRunmingConnectors,
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/Plugins/Plugins.js]:
```suggestion
    getRunningConnectors,
```

File: ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/Plugins/Plugins.js [hunk 2]
```diff
@@ -256,74 +265,91 @@ const Plugins = props => {
...
+
+  const handleError = async params => {
+    const newError = errors.push(params);
+    setErrors(newError);
+
+    const reverts = errors.filter(error => error.success === false);
+    if (reverts.length > 0) {
+      deleteType.current = true;
+    }
+    for (var revert of reverts) {
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/Plugins/Plugins.js]: `var` is troublesome, please you `let` instead unless you have a good reason for using it :)

File: ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/Topics/Topics.js [hunk 3]
```diff
@@ -53,7 +53,9 @@ const Topics = props => {
   } = useApi.useDeleteApi(`${URL.TOPIC_URL}`);
   const { waitApi } = useApi.useWaitApi();
 
-  const topicsUnderBrokerCluster = get(topics, 'data.result', []);
+  const topics = get(topicsRes, 'data.result', []).filter(
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/Topics/Topics.js]: This issue is listed in #2457. If you're fixing it here, please reference the issue number to here from that issue

**Comment by konekoya** [comment]
retry manager

File: ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/Topics/Topics.js (hunk 3 above)
Comment (wu87988622) [reply ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/Topics/Topics.js]: ok thanks

File: ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/Plugins/Plugins.js (hunk 2 above)
Comment (wu87988622) [reply ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/Plugins/Plugins.js]: oh sorry i forget, my fault

File: ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/Plugins/Plugins.js (hunk 1 above)
Comment (wu87988622) [reply ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/Plugins/Plugins.js]: thanks for check

**Comment by wu87988622** [comment]
retry manager-e2e

**Comment by wu87988622** [comment]
retry test/manager-e2e

**Comment by wu87988622** [comment]
retry manager-e2e

**Pull request merged**